File: backend/app/models/job.py
import uuid

from sqlalchemy import Text
from sqlalchemy import JSON
from sqlalchemy import String

from sqlalchemy.orm import Mapped
from sqlalchemy.orm import mapped_column

from sqlalchemy.dialects.postgresql import UUID

from pgvector.sqlalchemy import Vector

from app.db.base import Base


class Job(Base):

    __tablename__ = "jobs"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        primary_key=True,
        default=uuid.uuid4,
    )

    title: Mapped[str]

    company: Mapped[str] = mapped_column(
        String(255)
    )

    location: Mapped[str | None] = mapped_column(
        String(255),
        nullable=True,
    )

    description: Mapped[str] = mapped_column(
        Text
    )

    technologies: Mapped[list | None] = mapped_column(
        JSON,
        nullable=True,
    )

    embedding: Mapped[list[float] | None] = mapped_column(
        Vector(1536),
        nullable=True,
    )

File: frontend/src/types/job.ts
export interface Job {

    id: string;

    title: string;

    company: string;

    location: string | null;

    description: string;

    technologies: string[];

    match_score?: number;

}

export interface CrawledJob {
    title: string;
    company: string;
    location: string;
    department?: string;
    job_id: string;
    url: string;
}